src/changes.go

```diff
@@ -262,6 +262,7 @@ type changeSliceArg struct {
 	localParent, remoteParent string
 	dirList                   []*dirList
 	changeListPtr             *[]*Change
+	mtx                       *sync.Mutex
```

Comment:
Could you make this instead a non-pointer value and drop the extra the indirection ie
```go
mtx sync.Mutex
```